### DISCUSSION TITLE
Decompression error with PSD file

### DISCUSSION THREAD

**Discussion by softvisionfd** [ImageMagick/ImageMagick#6732] [Contribute]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagickDecompression error with PSD file
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When trying to convert (or identify) the attached PSD sample files (`Desktop2.psd` and `gfx_sliding_door.psd` in attached file `error.zip`
[error.zip](https://github.com/ImageMagick/ImageMagick/files/4556671/error.zip)
) the program throws a decompression error and is not able to handle the file at all (read file information or convert the image).

Other application, e.g. Gimp can read and display the PSD files.

### Steps to Reproduce
Call `magic.exe` from command line with 

`magick.exe identify -verbose Desktop2.psd` or 
`magick.exe" Desktop2.psd out.jpg`

### System Configuration
- ImageMagick version: `ImageMagick 7.0.10-1 Q8 x64 2020-03-16`
- Environment (Operating system, version and so on): Windows 10 x64, 
- Additional information:

**Comment by dlemstra** [comment]
The ImageMagick library is very strict and will report an error when the file is not correct. Can you show how the image looks in Gimp with a screenshot and attach the image here?

**Comment by softvisionfd** [comment]
Hi, 

I have opened both file in gimp (https://www.gimp.org/) and IrfanView (https://www.irfanview.com/). Both applications can read, show and export the PSD files. I've made a screenshot of each application and a saved (exported) JPEG file. Everything is in the attached ZIP file.

If you need further information, please let me know.

Regards
Bernd

[screenshots_and_export.zip](https://github.com/ImageMagick/ImageMagick/files/4567460/screenshots_and_export.zip)

**Comment by dlemstra** [comment]
The problem for both images appear to be that they have an invalid rowsize. Maybe we could add a flag `psd:ignore-row-size` in the future to allow reading of invalid images.
